### DISCUSSION TITLE
Real-time speaker diarization with Pyannote.audio: Adding multiple files to the pipeline

### DISCUSSION THREAD

**Discussion by uselessai** [pyannote/pyannote-audio#1356] [Q&A]
Hello,

I'm interested in using Pyannote.audio to analyze multiple audio files in real-time. Is it possible to add multiple files to the pipeline and have it analyze them without losing track of the speakers identified in previous files?

Ideally, I would like to continuously add new files to the pipeline as they become available, without having to concatenate them beforehand. Is this something that Pyannote.audio supports?

Thank you for your help!

**Comment by hbredin** [comment]
As of May 2023, pyannote.audio does not do real-time (did you mean streaming?) diarization.
See [diart](https://github.com/juanmc2005/diart) instead.

Regarding your actual question about tracking speakers over different files, the short answer is "no".
But, but, but, note that there is an ongoing PR that will probably get us closer to that feature.
See https://github.com/pyannote/pyannote-audio/pull/1346
